sdk/storage/storage-queue/test/utils/recorder.ts

```diff
@@ -0,0 +1,310 @@
+      file.write(fixture + "\n");
+    }
+    file.end();
+
+    nock.recorder.clear();
+    nock.restore();
+  }
+}
+
+class NiseRecorder extends Recorder {
+  private readonly sasQueryParameters = ["se", "sig", "sp", "spr", "srt", "ss", "st", "sv"];
+  private recordings: any[] = [];
+
+  constructor(testHierarchy: string, testTitle: string) {
+    super("browsers", testHierarchy, testTitle, "json");
+  }
+
+  private async recordRequest(request: any, data: any): Promise<void> {
+    const responseHeaders: any = {};
+    const responseHeadersPairs = request.getAllResponseHeaders().split("\r\n");
+    for (const pair of responseHeadersPairs) {
+      const [key, value] = pair.split(": ");
+      responseHeaders[key] = value;
+    }
+
+    // We're not storing SAS Query Parameters because they may contain sensitive information
+    // We're ignoring the "_" parameter as well because it's not being added by our code
+    // More info on "_": https://stackoverflow.com/questions/3687729/who-add-single-underscore-query-parameter
+    const parsedUrl = queryString.parseUrl(request.url);
+    const query: any = {};
+    for (const param in parsedUrl.query) {
+      if (!this.sasQueryParameters.includes(param) && param !== "_") {
+        query[param] = parsedUrl.query[param];
+      }
+    }
+
+    this.recordings.push({
+      method: request.method,
+      url: parsedUrl.url,
+      query: query,
+      requestBody: (data instanceof Blob) ? await blobToString(data) : data,
+      status: request.status,
+      response: (request.response instanceof Blob) ? await blobToString(request.response) : request.response,
+      responseHeaders: responseHeaders
+    });
+  }
+
+  // We're not matching request headers
+  private matchRequest(recording: any, request: any): boolean {
+    for (const param in recording.query) {
+      if (recording.query[param] !== request.query[param]) {
+        return false;
+      }
+    }
+
+    for (const param in request.query) {
+      if (recording.query[param] === undefined && !this.sasQueryParameters.includes(param) && param !== "_") {
+        return false;
+      }
+    }
+
+    return (
+      recording.method === request.method &&
+      recording.url === request.url &&
+      recording.requestBody === request.requestBody
+    );
+  }
+
+  public record(): void {
+    const self = this;
+    const xhr = nise.fakeXhr.useFakeXMLHttpRequest();
+
+    xhr.useFilters = true;
+    xhr.addFilter(() => true);
+
+    xhr.onCreate = function(req: any) {
+      const reqOpen = req.open;
+      req.open = function() {
+        reqOpen.apply(req, arguments);
+
+        const reqSend = req.send;
+        req.send = function(data: any) {
+          const reqStateChange = req.onreadystatechange;
+          req.onreadystatechange = function() {
+            if (req.readyState === 4) {
+              self.recordRequest(req, data);
+            }
+            reqStateChange && reqStateChange.apply(null, arguments);
```

Comment:
@kinelski 
Looks like an unused expression

Reply:
I added an 'if' statement to make this piece of code clearer.